rosbag2_cpp/include/rosbag2_cpp/bag_events.hpp

```diff
@@ -0,0 +1,131 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_CPP__BAG_EVENTS_HPP_
+#define ROSBAG2_CPP__BAG_EVENTS_HPP_
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rclcpp/function_traits.hpp"
+#include "rosbag2_cpp/visibility_control.hpp"
+
+namespace rosbag2_cpp
+{
+
```

Comment:
Would some comments in this file explaining what the events are and how to implement them be useful? I'm just thinking of future contributors to `rosbag2` that may not have been privy to the discussions on this. Comments may help to reduce onboarding time and lessen potential confusion. Maybe.

Reply:
Good catch! That file was completely devoid of comments. I've documented the entities in it in ad46aaa. Please take a look.